tests/50federation/40devicelists.pl

```diff
@@ -338,17 +334,7 @@
          );
       })->then( sub {
          log_if_fail "left room; now rejoining";
-         my $iter = 0;
-         retry_until_success {
-            $iter++;
-            matrix_join_room( $user, $room->room_id,
-               server_name => $inbound_server->server_name,
-            )->on_fail( sub {
-               my ( $exc ) = @_;
-               chomp $exc;
-               log_if_fail "Room join iteration $iter failed: $exc";
-            });
```

Comment:
isn't this loop important?

Reply:
At least I'm not seeing why it should be important.
We're leaving the room synced, wait for the federated server to receive the leave and re-join the room.

Reply:
hrm you're probably right!